src/utils/reader_chunks.ts

```diff
@@ -102,52 +102,52 @@ class BufferChunksReader {
 
   readUInt8() {
     const buf = this.read(1);
-    return buf.readUInt8();
+    return buf?.readUInt8();
```

Comment:
@kevinejohn maybe all of these read methods should throw an Error if `buf === undefined`. That way the return type of all these methods would be a defined type.

Reply:
lets make it throw if it is `undefined`